Client/lib/PONAPI/Client/UA.pm

```diff
@@ -0,0 +1,45 @@
+package PONAPI::Client::UA;
+
+use Moose;
+
+has type => (
+    is      => 'ro',
+    isa     => 'Str',
+    default => sub { 'Hijk' },
+);
+
+has ua_args => (
+    is      => 'ro',
+    isa     => 'ArrayRef',
+    default => sub { +[] },
+);
+
+sub BUILDARGS {
+    my $class = shift;
+    my %args = @_ == 1 ? %{ $_[0] } : @_;
+
+    if ( $args{ua_args} ) {
+        my %ua_args = @{ $args{ua_args} };
+        if ( exists $ua_args{type} ) {
+            $args{type} = $ua_args{type};
+            die "unsupported UA type: $args{type}\n"
+                unless grep { $_ eq $args{type} } qw< Hijk YAHC >;
+        }
+    }
+
+    return \%args;
+}
+
+
+sub build {
+    my $self = shift;
+
+    return (
+        $self->type eq 'Hijk' ? PONAPI::Client::UA::Hijk->new( @{ $self->ua_args } ) :
```

Comment:
yeah, as said, I don't like this BUILDARGS at all... this PR is more of a draft and I would like to remove this part.

This whole part has nothing to do with caller passing an object - this is only called in the builder, so only in the case such object wasn't passed.
This solution is attempting to also support the ability of not passing an object but just declaring the type of UA and let the constructor deal with it. 

to simplify it, I can separate `ua_args` from `ua_type` to support this feature with the BUILDARGS ugliness.